common/lib/xmodule/xmodule/contentstore/content.py

```diff
@@ -330,17 +334,29 @@ def generate_thumbnail(self, content, tempfile_path=None, dimensions=None):
         pixels. It defaults to None.
         """
         thumbnail_content = None
+        is_svg = content.content_type == 'image/svg+xml'
         # use a naming convention to associate originals with the thumbnail
         thumbnail_name = StaticContent.generate_thumbnail_name(
-            content.location.name, dimensions=dimensions
+            content.location.name, dimensions=dimensions, extension='.svg' if is_svg else None
         )
         thumbnail_file_location = StaticContent.compute_location(
             content.location.course_key, thumbnail_name, is_thumbnail=True
         )
 
         # if we're uploading an image, then let's generate a thumbnail so that we can
         # serve it up when needed without having to rescale on the fly
-        if content.content_type is not None and content.content_type.split('/')[0] == 'image':
+        if is_svg:
+            # for svg simply store the provided svg file, since vector graphics should be good enough
+            # for downscaling client-side
+            if tempfile_path is None:
+                thumbnail_file = StringIO.StringIO(content.data)
+            else:
+                with open(tempfile_path) as f:
```

Comment:
This will take care of the closing (even in the event of an error), but if an error is thrown, the rest of the code will bomb. How about moving the try/except handling that starts on line 360 so that it also contains this new section of code?